packages/expect/src/to_throw_matchers.js

```diff
@@ -28,21 +28,24 @@ export const createMatcher = (matcherName: string, fromPromise?: boolean) => (
   const value = expected;
   let error;
 
-  if (fromPromise) {
+  if (fromPromise && actual instanceof Error) {
```

Comment:
Is the `instanceof` check safe? Ref #2549.

I guess it is seeing as tests pass.

/cc @mjesun

Reply:
No it isn't because there will be one instance of the Error class in both the parent and child contexts. We can only use duck typing.

Reply:
Thanks for catching that. Is `typeof actual.name === 'string' && typeof actual.message === 'string'` enough instead of `actual instanceof Error` condition?

Reply:
Or maybe use `actual.toString && typeof actual.toString === 'function' && actual.toString() === Error.prototype.toString.call(actual)`?